contributors/devel/sig-testing/testing-strategy.md

```diff
@@ -0,0 +1,99 @@
+## Defining a Robust Testing Strategy
+
+This document outlines a testing strategy for Kubernetes features based on the **testing pyramid**, with considerations on the existing CI system's characteristics.
+
+### The Testing Pyramid
+
+Prioritize tests based on the testing pyramid, refer to the [Testing Guide](./testing.md):
+
+- **Unit Tests:**  The foundation. Fast, isolated, and cover individual components.
+- **Integration Tests:** Verify interactions between components within your subsystem.
+- **E2E Tests:**  Test the entire system, including interactions with external dependencies. These are the most expensive and prone to flakiness.
+
+### CI Job Types
+
+The Kubernetes job uses [prow](https://prow.k8s.io) to implement the CI system. We can differentiate between different types of jobs:
+
+- **Presubmit:** Runs before code is merged.
+    - **Blocking:** Prevents merging if tests fail. Use cautiously due to potential project-wide impact. We aim to have a very high bar on these jobs and ask for proof
+    of stability, reliability and performance.
+    - **Non-Blocking/Informational:** Provides feedback without blocking merges.
+- **Periodic:** Runs at scheduled intervals. Ideal for monitoring trends and catching regressions.
+- **Postsubmit:** Runs after code is merged. Useful for building artifacts.
```

Comment:
nit: I'd put postsubmit before periodic. imo it lines up with where tests ideally go (presubmit if they're fast enough, postsubmit if they're not but we want the info from every PR (and also for stuff that only makes sense once the code's submitted), and periodic if tests are very expensive).